metrics/web/test/auth/domain/usercases/google_sign_in_usecase_test.dart

```diff
@@ -9,6 +11,23 @@ import '../../../test_utils/user_repository_mock.dart';
 void main() {
   group("GoogleSignInUseCase", () {
     final repository = UserRepositoryMock();
+    const email = 'email';
+
+    setUp(() {
+      when(repository.getGoogleSignInCredentials()).thenAnswer(
+        (_) => Future.value(
+          AuthCredentials(
+            email: email,
+            accessToken: 'accessToken',
+            idToken: 'idToken',
+          ),
+        ),
+      );
+
+      when(repository.validateEmailDomain(any)).thenAnswer(
+        (_) => Future.value(EmailDomainValidationResult(isValid: true)),
+      );
```

Comment:
I suppose we should avoid such dependencies for test cases. Let's move this to the function that will stand for multiple mocks such as: 
```dart
PostExpectation<Future<EmailDomainValidationResult>> whenValidateEmailDomain({String email, String accessToken: 'accessToken', String idToken: 'idToken'}) {}
```

This helps to make tests more clear and avoid hidden dependencies.